src/View/Widget/DateTimeWidget.php

```diff
@@ -303,11 +325,17 @@ protected function _adjustValue($value, $options)
      */
     protected function _yearSelect($options, $context)
     {
+        $date = new Time(null, $options['localization']['timezone']);
+
+        if (is_string($options['val'] && !empty($options['val']))) {
+            $date->year($options['val']);
+        }
+
         $options += [
             'name' => '',
             'val' => null,
-            'start' => date('Y', strtotime('-5 years')),
-            'end' => date('Y', strtotime('+5 years')),
+            'start' => $date->subYears(5)->I18nFormat("yyyy", null, $options['localization']['baseLocale']),
+            'end' => $date->addYears(10)->I18nFormat("yyyy", null, $options['localization']['baseLocale']),
             'order' => 'desc',
```

Comment:
Is the baseLocale key always set? In the string parsing code it gets set to null.